src/libraries/Microsoft.PowerFx.LanguageServerProtocol/LanguageServer/LanguageServer.cs

```diff
@@ -493,7 +493,12 @@ private void HandleSemanticTokens<T>(string id, T semanticTokensParams, string m
 
             var tokenTypesToSkip = ParseTokenTypesToSkipParam(queryParams?.Get("tokenTypesToSkip"));
             var scope = _scopeFactory.GetOrCreateInstance(semanticTokensParams.TextDocument.Uri);
-            var result = scope.Check(expression);
+            var result = scope?.Check(expression);
+            if (scope == null)
+            {
+                SendEmptySemanticTokensResponse(id);
+                return;
+            }
 
```

Comment:
You can move this check to before the previous line, and after that you won't need to use the `?` null-propagating operator in the Check call.

Reply:
This is a nice catch. I meant to check for result == null and not just scope.